orchestrator/cli/migrate_workflows.py

```diff
@@ -0,0 +1,230 @@
+            headers="keys",
+            showindex=itertools.count(1),
+        ),
+        end="\n\n",
+    )
+    return state
+
+
+def delete_dangling_workflows(workflows: List[WorkflowTable], state: dict) -> dict:
+    print_fmt(
+        "\nThe following workflows were found in the database that do not have a corresponding LazyWorkflowInstance:\n"
+    )
+    # All workflow products have the same product_type, so we take the first product
+    # Workflows without a product  will not be listed
+    _print_workflows_table(workflows)
+    should_delete_dangling_workflows = (
+        get_user_input("Do you wish to delete all dangling workflows from the database? [y/n]: ", "n").lower() == "y"
+    )
+
+    if not should_delete_dangling_workflows:
+        _noqa_print("Cancelling")
+        return state
+
+    already_used_workflows = {wf["name"] for wf in state["workflows_to_add"] + state["workflows_to_delete"]}
+    keys = ["name", "target", "description", "product_type"]
+    items = [
+        {
+            "name": wf.name,
+            "target": wf.target,
+            "description": wf.description,
+            "product_type": wf.products[0].product_type,
+        }
+        for k, wf in zip(keys, workflows)
+        if wf.products and wf.name not in already_used_workflows
+    ]
+    return {**state, "workflows_to_delete": [*state["workflows_to_delete"], *items]}
+
+
+def create_workflows_migration_wizard() -> Tuple[List[dict], List[dict]]:
+    """Create tuple with lists for workflows to add and delete.
+
+    Returns tuple:
+        - list of workflow items to add in the migration
+        - list of workflow items to delete in the migration
+    """
+    database_workflows = WorkflowTable.query
+    registered_workflows = orchestrator.workflows.ALL_WORKFLOWS
+    system_workflow_names = {wf.name for wf in database_workflows if wf.target == Target.SYSTEM}
+    registered_non_system_workflows = {k: v for k, v in registered_workflows.items() if k not in system_workflow_names}
+
+    unregistered_workflows = [wf for wf in database_workflows if wf.name not in registered_workflows.keys()]
+
+    # Main menu loop
+    state = {"workflows_to_add": [], "workflows_to_delete": [], "done": False}
+    while not state["done"]:
+        print_fmt("\nWhat do you want to do?\n", flags=[COLOR.UNDERLINE, COLOR.BOLD])
+        choice_fn = _prompt_user_menu(
+            [
+                ("Add workflow to database", lambda s: _add_workflow(registered_non_system_workflows, s)),
+                ("Delete workflow from database", lambda s: _delete_workflow(database_workflows, s)),
+                (
+                    "Delete unregistered workflows from database",
+                    lambda s: delete_dangling_workflows(unregistered_workflows, s),
+                ),
+                ("Finish and create migration file", lambda s: {**s, "done": True, "abort": False}),
+                ("Show current diff", lambda s: _show_state(s)),
```

Comment:
this is very nice, I would prefer always seeing the diff above the question `What do you want to do?`